src/stacks-hierarchy/Root_LoginConfirmCodeScreen.tsx

```diff
@@ -20,35 +23,34 @@ import {StyleSheet, useTheme} from '@atb/theme';
 import {getStaticColor, StaticColorByType} from '@atb/theme/colors';
 import {RootStackScreenProps} from '@atb/stacks-hierarchy/navigation-types';
 import {PressableOpacity} from '@atb/components/pressable-opacity';
-import {useCompleteOnboardingAndEnterApp} from '@atb/utils/use-complete-onboarding-and-enter-app';
+import {useAppState} from '@atb/AppContext';
 
 const themeColor: StaticColorByType<'background'> = 'background_accent_0';
 
 type Props = RootStackScreenProps<'Root_LoginConfirmCodeScreen'>;
 
 export const Root_LoginConfirmCodeScreen = ({route}: Props) => {
-  const {phoneNumber, afterLogin} = route.params;
+  const {phoneNumber} = route.params;
   const {t} = useTranslation();
   const styles = useStyles();
   const {themeName} = useTheme();
-  const {authenticationType, confirmCode, signInWithPhoneNumber} =
-    useAuthState();
+  const {confirmCode, signInWithPhoneNumber} = useAuthState();
   const [code, setCode] = useState('');
   const [error, setError] = useState<
     ConfirmationErrorCode | PhoneSignInErrorCode
   >();
   const [isLoading, setIsLoading] = useState(false);
   const focusRef = useFocusOnLoad();
-  const completeOnboardingAndEnterApp = useCompleteOnboardingAndEnterApp();
+  const {completeOnboarding} = useAppState();
 
   const onLogin = async () => {
     setIsLoading(true);
+    completeOnboarding();
```

Comment:
So if the user was onboarded, but code fails or wrong / expired / etc, does it end as completed but not fully completed because of the error?

Reply:
@jorelosorio Yes. If the error code fails, and the app is restarted, it will continue as if you chose continue without logging in. This is not great obviously, but not a catastrophe either I guess 😅 
Ideally we would set the `onboarded` flag when we know there was no error and before entering the app. (What we want to avoid is entering the app before the onboarded flag has been set.)

Reply:
> So if the user was onboarded, but code fails or wrong / expired / etc, does it end as completed but not fully completed because of the error?

Note that the function `completeOnboarding` only sets the onboarded flag to `true` in local storage, it doesn't navigate or something. So if the the user types wrong code, he stays in the onboarding with an error msg saying "wrong code", and can try again or request a new one.